test/test_mkldnn_fusion.py

```diff
@@ -302,6 +341,57 @@ def forward(self, x, other):
 
                         self.assertEqual(ref, fused)
 
+    def test_conv_binary_fusion_ops_with_param(self):
+        class M(nn.Module):
+            def __init__(self, binary_fn, dim, in_channels, out_channels, dilation, groups, bias, **kwargs):
+                super(M, self).__init__()
+                self.conv = CONV_MODULES[dim](in_channels, out_channels, dilation=dilation, groups=groups, bias=bias, **kwargs)
+                self.binary = binary_fn
+
+            def forward(self, x, other):
+                x = self.conv(x)
+                x = self.binary(x, other)
+                return x
+
+        input_shapes = {2: (112, 112), 3: (55, 55, 55)}
+        for pointwise_name, pointwise_fn in self._binary_list().items():
+            for dim in [2, 3]:
+                channels_last = torch.channels_last if dim == 2 else torch.channels_last_3d
+                options = itertools.product([False, True], [True, False], [1, 2], [1, 4], [torch.contiguous_format, channels_last])
+                for fuse_relu, bias, dilation, groups, memory_format in options:
+                    oC = 32 * groups
+                    iC = 3 * groups
+                    x_shape = (1, iC) + input_shapes[dim]
+                    x = torch.randn(x_shape, dtype=torch.float32).to(memory_format=memory_format)
+                    mod = M(pointwise_fn, dim, iC, oC, dilation, groups, bias, kernel_size=3)
+                    mod = mod.to(memory_format=memory_format).eval()
+                    other = torch.randn_like(mod.conv(x))
+                    with torch.no_grad():
+                        ref = mod(x, other)
+                        unary_attr = None
+                        if fuse_relu:
+                            ref.relu_()
+                            unary_attr = "relu"
+                        attr = pointwise_name
+                        conv_param = torch.ops.mkldnn._create_conv_param_binary(
+                            x, other, mod.conv.weight, mod.conv.bias, mod.conv.padding, mod.conv.stride, mod.conv.dilation,
+                            mod.conv.groups, attr, None, unary_attr, [], None
+                        )
+                        fused = torch.ops.mkldnn._convolution_pointwise(
+                            x, other, mod.conv.weight, mod.conv.bias, mod.conv.padding, mod.conv.stride, mod.conv.dilation,
+                            mod.conv.groups, attr, None, unary_attr, [], None, conv_param
+                        )
+                        torch.mkldnn_delete_param(conv_param, 0)
```

Comment:
IMO it would be cleaner if we could tie the garbage collection of these parameter objects to the python `conv_param` object, rather than exposing C++ memory management to python.  Python normally uses reference counting and you can run code in the destructor when the reference count goes to zero.

Reply:
Yeah, here I just explictly use the delete_param function in test case to practice this function. In the real code in torch/_inductor/codegen/wrapper.py, these parameter objects are tied to mkldnn_param_cache python object, and will be auto deconstructed within the __del__. This mkldnn_delete_param will not be called directly by any other places explictly.

Reply:
This logic doesn't seem inductor-specific.  We should try to have a sane API at the `torch.ops.*` boundary.